gadget/kcmdline.go

```diff
@@ -0,0 +1,62 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2023 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package gadget
+
+import (
+	"fmt"
+
+	"github.com/snapcore/snapd/osutil"
+)
+
+type kargKey struct{ par, val string }
+type kernelArgsSet map[kargKey]bool
+
+// CheckCmdlineAllowed returns an error if an argument from cmdline is
+// not on a list of allowed kernel arguments. A wild card ('*') can be
+// used in the allow list for the values.
+func CheckCmdlineAllowed(cmdline string, allowedSl []osutil.KernelArgument) error {
+	// Set of allowed arguments
+	allowed := kernelArgsSet{}
+	wildcards := map[string]bool{}
+	for _, p := range allowedSl {
+		if p.Value == "*" && !p.Quoted {
+			// Currently only allowed globbing
```

Comment:
Just to double check my understanding - this is not used right now, i.e. we do not allow quoting on the higher levels ?

Reply:
I'm not sure if I understand the question :slightly_smiling_face:, but to summarize using globbing characters (with the exception of having a single `*` value) is not allowed unless we are quoting so we can use more general globbing in the future without breaking older gadgets. If we are quoting we interpret the characters literally. And the only globbing that we support atm is having `param=*`.

Reply:
Thanks! And sorry that I expressed myself a bit clumsy here.